include/mxnet/c_api.h

```diff
@@ -516,6 +516,19 @@ MXNET_DLL int MXImperativeInvoke(AtomicSymbolCreator creator,
                                  const char **param_keys,
                                  const char **param_vals);
 
+MXNET_DLL int MXAutogradStartRecording();
+MXNET_DLL int MXAutogradStopRecording();
+MXNET_DLL int MXAutogradMarkForRecord(mx_uint num_arr,
+                                      NDArrayHandle *arrays);
+MXNET_DLL int MXAutogradUnMarkForRecord(mx_uint num_arr,
```

Comment:
merge these two to one.

side note: why do we need to mark individual ndarrays?